src/wasm.h

```diff
@@ -786,7 +786,7 @@ class Module {
   Function* getFunctionOrNull(Name name);
   Global* getGlobalOrNull(Name name);
 
-  void addFunctionType(FunctionType* curr);
+  FunctionType* addFunctionType(std::unique_ptr<FunctionType> curr);
```

Comment:
I agree this is better. For consistency we should do the others as well. I wouldn't be opposed to separate PRs, during which we are not consistent in the middle, but only if we're sure we can get to the other PRs quickly.

Reply:
Sure, I can send following PRs changing other methods. Do you mean the methods taking raw pointers only in this class?

Here I wanted to send the set of minimal changes to get the feedback about the proposed changes.

Should I change the `curr` argument name? I don't think it is a good name here.

Reply:
It might make sense to split up into multiple PRs, good point.

It might be best to wait for #1678 to land first, before any of this, though - it's another widespread change and the two may have conflicts.

About `curr`, it's short for "current" when there isn't a better name. I don't think it's worth changing as it's very widespread in the codebase.

Reply:
I applied requested changes. If you want me to wait for some monster PR touching 315 files to be merged... it does not sound reasonable.